basho/cloudformation-riak
# New Nodes Join Cluster Properly When ASG Size is Increased

Changes the way the node that all new nodes join and the node that plans/commits the cluster change are chosen.

Before, they were chosen using the first and last elements of a sorted list of internal ip addresses.  With this change, the sort is now on based on launch_time and instance-id which assures that one of the most recently launched instances will attempt to plan and commit the cluster changes.

To test it out using the new AWS cli:

```
aws autoscaling update-auto-scaling-group --auto-scaling-group-name \ 
<asg_name> --max-size NEW_MAX_SIZE
aws autoscaling set-desired-capacity --auto-scaling-group-name <asg_name> \
--desired-capacity DESIRED_CAPACITY
```

after a few minutes, ssh to one of the boxes in the cluster and you can see the new nodes have been added to the cluster.

Fixes issue #3
